termennetwerk.ts

```diff
@@ -51,22 +53,20 @@ export async function getUriOfSearchTerm(sources: string[], searchTerm: string):
     const cacheKey = searchTerm.toLowerCase();
     if (cache[cacheKey]) return cache[cacheKey];
     const response: GraphQlResponse = await client.request(graphQlQuery, { sources, query: searchTerm });
-    if (!Array.isArray(response.terms) ||
-        response.terms[0].result.terms.length !== 1 ||
-        !response.terms[0].result ||
-        !response.terms[0].result.terms ||
-        !Array.isArray(response.terms[0].result.terms)) {
+    const result = response.terms[0].result;
+    if (!Array.isArray(result.terms) ||   // terms[] does not exists when an error occurs like a timeout
```

Comment:
Discriminating on the result type (`result.__typename === 'Terms'`) is cleaner and less prone to break than checking the `terms` property.

Reply:
I agree and I tried to that but it failed with:
` Property '__typename' does not exist on type '{ terms: { uri: string; prefLabel: string[]; altLabel: string[]; }[]; message: string; }'.` 
So it unclear to me how to do this properly.

Reply:
Ah thanks, that helps!